graql/reasoner/atom/binary/AttributeAtom.java

```diff
@@ -216,7 +216,9 @@ public final boolean equals(Object obj) {
         AttributeAtom a2 = (AttributeAtom) obj;
         return Objects.equals(this.getTypeLabel(), a2.getTypeLabel())
                 && this.getVarName().equals(a2.getVarName())
+                && this.getAttributeVariable().equals(a2.getAttributeVariable())
```

Comment:
We might want to update the hashCode as well. Currently we are not breaking the hashCode/equals contract (equal attributes have equal hashcodes, non-equal attributes can have equal hashcodes), however we could avoid doing extra `equals` check for cases when attribute values are different.